pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -129,6 +129,26 @@ public SuccessResponse addSchema(FormDataMultiPart multiPart) {
     return addOrUpdateSchema(null, multiPart);
   }
 
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/schemas/validate")
+  @ApiOperation(value = "Validate schema", notes = "This API returns the table config that matches the one you get "
+      + "from 'GET /schema/{schemaName}'. This allows us to validate schema before apply.")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Successfully validated schema"),
+      @ApiResponse(code = 400, message = "Missing or invalid request body"),
+      @ApiResponse(code = 500, message = "Internal error")})
+  public String validateSchema (FormDataMultiPart multiPart) throws Exception {
+    try {
+      Schema schema = getSchemaFromMultiPart(multiPart);
+      if (!schema.validate(LOGGER)) {
+        throw new ControllerApplicationException(LOGGER, "Invalid schema", Response.Status.BAD_REQUEST);
```

Comment:
I suggest changing the message to: "Invalid schema. Check controller logs"

The logs will identify which field had an error, if it is possible to do so.

Reply:
fixed.